packages/sdk/src/evm/functions/contract/index.ts

```diff
@@ -0,0 +1,102 @@
+import { getCompositePluginABI } from "../../common/plugin";
+import { PREBUILT_CONTRACTS_MAP } from "../../contracts";
+import { SmartContract } from "../../contracts/smart-contract";
+import { ContractPublisher } from "../../core/classes/contract-publisher";
+import { getSignerAndProvider } from "../../core/classes/rpc-connection-handler";
+import { ValidContractInstance } from "../../core/types";
+import { AbiSchema } from "../../schema/contracts/custom";
+import {
+  cacheContract,
+  getCachedContract,
+  getCachedStorage,
+  inContractCache,
+} from "../core/cache";
+import { GetContractFromAbiParams, GetContractParams } from "./types";
+import { resolveContractType } from "./utils";
+
+export async function getContractFromAbi(
+  params: GetContractFromAbiParams,
+): Promise<ValidContractInstance> {
+  if (inContractCache(params.address)) {
+    return getCachedContract(params.address);
+  }
+
+  // TODO: Remove support for SDKOptions
```

Comment:
what do we still need it for in here?

Reply:
For temporary compatibility since we take thirweb/alchemy/infura API keys in here which works with our chains RPC URL setup

Reply:
As well as other small things like read only RPC which the provider functions use (for now)